docs/third-party-developers/extensibility/data-store/cart.md

```diff
@@ -0,0 +1,378 @@
+- `shippingAddress` - object containing the shipping address
+- `billingAddress` - object containing the billing address
+- `items` - array containing the cart items
+- `itemsCount` - number containing total number of items in the cart
+- `itemsWeight` - number containing the total weight of items in the cart
+- `crossSells` -  array containing the cross sells
+- `needsPayment` - boolean indicating if the cart needs payment
+- `needsShipping`- boolean indicating if the cart needs shipping
+- `hasCalculatedShipping`- boolean indicating if the cart has calculated shipping
+- `fees`- array containing the cart fees
+- `totals`- object containing the cart totals
+- `errors`- array containing the cart errors
+- `paymentRequirements`- object containing the payment requirements
+- `extensions`- object containing the extensions data
+
+#### _Example_
+
+```js
+const store = select( 'wc/store/cart' );
+const cartData = store.getCartData();
+```
+
+### getCustomerData
+
+Returns the shipping and billing address from the state
+
+#### _Returns_
+
+`object`- The current shipping and billing address. This will be an object with the following keys:
+
+- `shippingAddress`- Object containing the shipping address
+- `billingAddress`- Object containing the billing address
+
+
+#### _Example_
+
+```js
+const store = select( 'wc/store/cart' );
+const customerData = store.getCustomerData();
+```
+
+### getShippingRates
+
+Returns the shipping rates from the state
+
+#### _Returns_
+
+`array`- array containing the shipping rates.
+
+#### _Example_
+
+```js
+const store = select( 'wc/store/cart' );
+const shippingRates = store.getShippingRates();
+```
+
+### getNeedsShipping
+
+Queries whether the cart needs shipping
+
+#### _Returns_
+
+`boolean`- True if the cart needs shipping
+
+#### _Example_
+
+```js
+const store = select( 'wc/store/cart' );
+const needsShipping = store.getNeedsShipping();
+```
+
+### getHasCalculatedShipping
+
+Queries whether the cart shipping has been calculated.
+
+#### _Returns_
+
+`boolean`- True if the shipping has been calculated.
+
+#### _Example_
+
+```js
+const store = select( 'wc/store/cart' );
+const hasCalculatedShipping = store.getHasCalculatedShipping();
+```
+
+### getCartTotals
+
+Returns the cart totals from state.
+
+#### _Returns_
+
+`object`- The current cart totals. This will be an object with the following keys:
+
+- `total_items`- string containing the total items
+- `total_items_tax`- string containing the total items tax
+- `total_fees`- string containing the total fees
+- `total_fees_tax`- string containing the total fees tax
+- `total_discount`- string containing the total discount
+- `total_discount_tax`- string containing the total discount tax
+- `total_shipping`- string containing the total shipping
+- `total_shipping_tax`- string containing the total shipping tax
+- `total_tax`- string containing the total tax
+- `total_price`- string containing the total price
+- `tax_lines`- array containing the tax lines
```

Comment:
I think the language here could be a little clearer, for example `total_items` is the sum total of items in the cart without the tax or shipping applied to that item. Please could you try to improve this?